### DISCUSSION TITLE
how do int_printing settings for order work?

### DISCUSSION THREAD

**Discussion by smichr** [sympy/sympy#23773] [General]
The docs don't show an example that shows anything different for a given expression when changing the order flag. And when I use the same expression but change the order flag, I don't see a difference. How is this suppose to work?
```
 def do(e):
...  init_printing(order='lex')
...  print(e)
...  init_printing(order='grlex')
...  print(e)
...  init_printing(order='grevlex')
...  print(e)
... 
 do(x**2+x*y+y**2+z+3)
x**2 + x*y + y**2 + z + 3
x**2 + x*y + y**2 + z + 3
x**2 + x*y + y**2 + z + 3
```

**Comment by asmeurer** [comment]
This stuff is under-documented, and probably buggy as well. I think what you write is supposed to work, but it doesn't seem to.

**Comment by asmeurer** [comment]
Actually I think the problem is that your polynomial is ordered the same way in all three orders. Try looking at `x**2 + y**2 + z**2 + x*y + x*z + y*z + x + y + z + 1`. See https://en.wikipedia.org/wiki/Monomial_order.

It is still not documented very well, though. Also I noticed that incorrect orders are not checked in init_printing, meaning you don't get an error for a typo until you actually try to print something.
